drivers/pinctrl/pinctrl_xmc4xxx.c

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2022 Schlumberger
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gpio/gpio_xmc4xxx.h>
```

Comment:
why is this needed? please keep gpio/pinctrl independent

Reply:
> why is this needed? please keep gpio/pinctrl independent

I was following the example in pinctrl_stm32.c.

I used the gpio data structure to get the gpio port base address.
But I can also compute it from the pinctrl base address. Do you prefer this approach?

Reply:
that's the only case it's this way, mainly for historical reasons and it should be fixed.